model/task/db.go

```diff
@@ -2112,7 +2144,13 @@ func GetTasksByVersion(ctx context.Context, versionID string, opts GetTasksByVer
 
 	env := evergreen.GetEnvironment()
 	cursor, err := env.DB().Collection(Collection).Aggregate(ctx, pipeline)
+
 	if err != nil {
+		// If the pipeline stage is too large we should use the slow annotations lookup
+		if db.IsErrorCode(err, db.FACET_PIPELINE_STAGE_TOO_LARGE_CODE) {
+			opts.UseSlowAnnotationsLookup = true
+			return GetTasksByVersion(ctx, versionID, opts)
```

Comment:
Since this is a recursive call, can we make sure we don't have a chance of infinite recursion by checking if `UseSlowAnnotationsLookup` was already set for this `GetTasksByVersion` call?